core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,468 @@
+          collection, 'non_existent')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+    });
+
+    it('should handle shifting the first node', function() {
+      var collection = createLinearCollection();
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+      expect(CollectionLinearizerService.shiftNodeRight(
+          collection, 'exp_id0')).toBe(true);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([secondCollectionNode, firstCollectionNode,
+                  thirdCollectionNode]);
+    });
+
+    it('should handle shifting the last node', function() {
+      var collection = createLinearCollection();
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+      expect(CollectionLinearizerService.shiftNodeRight(
+          collection, 'exp_id2')).toBe(true);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+    });
+
+    it('should correctly handle shifting middle node', function() {
+      var collection = createLinearCollection();
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+
+      expect(CollectionLinearizerService.shiftNodeRight(
+          collection, 'exp_id1')).toBe(true);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, thirdCollectionNode,
+                secondCollectionNode]);
+    });
+  });
+
+  describe('getNextExplorationIds()', function() {
+    it('should return no exploration ids for last node in linear collection',
+        function() {
+      var collection = createLinearCollection();
+      expect(CollectionLinearizerService.getNextExplorationIds(collection,
+          ['exp_id0', 'exp_id1', 'exp_id2'])).toEqual([]);
+    });
+
+    it('should correctly return the next exploration id for linear collection',
+        function() {
+      var collection = createLinearCollection();
+      expect(CollectionLinearizerService.getNextExplorationIds(collection,
+          ['exp_id0', 'exp_id1'])).toEqual(['exp_id2']);
+    });
+
+    it('should return no ids for the last node in non-linear collection',
```

Comment:
Ah, you're right. Good catch and good idea on writing the tests. I forgot about that comment. :)